samples/snippets/csharp/language-reference/keywords/numeric-literals.cs

```diff
@@ -0,0 +1,314 @@
+using System;
+
+public class Example
+{
+   public static void Main()
+   {
+       Console.WriteLine("\nByte Assignments:");
+       AssignByte();
+       Console.WriteLine("\nByte Assignments with Separator:");
+       AssignByteWithSeparator();
+
+       Console.WriteLine("\nShort Assignments:");
+       AssignShort();
+       Console.WriteLine("\nShort Assignments with Separator:");
+       AssignShortWithSeparator();
+
+       Console.WriteLine("\nInteger Assignments:");
+       AssignInteger();
+       Console.WriteLine("\nInteger Assignments with Separator:");
+       AssignIntegerWithSeparator();
+
+       Console.WriteLine("\nLong Assignments:");
+       AssignLong();
+       Console.WriteLine("\nLong Assignments with Separator:");
+       AssignLongWithSeparator();
+
+       Console.WriteLine("\nSigned Byte Assignments:");
+       AssignSByte();
+       Console.WriteLine("\nSigned Byte Assignments with Separator:");
+       AssignSByteWithSeparator();
+
+       Console.WriteLine("\nUShort Assignments:");
+       AssignUShort();
+       Console.WriteLine("\nUShort Assignments with Separator:");
+       AssignUShortWithSeparator();
+
+       Console.WriteLine("\nUInteger Assignments:");
+       AssignUInteger();
+       Console.WriteLine("\nUInteger Assignments with Separator:");
+       AssignUIntegerWithSeparator();
+
+       Console.WriteLine("\nULong Assignments:");
+       AssignULong();
+       Console.WriteLine("\nULong Assignments with Separator:");
+       AssignULongWithSeparator();
+   }
+
+   private static void AssignByte()
+   {
+      // <SnippetByte>
+      byte byteValue1 = 201;
+      Console.WriteLine(byteValue1);
+      
+      byte byteValue2 = 0x00C9;
+      Console.WriteLine(byteValue2);
+      
+      byte byteValue3 = 0b11001001;
+      Console.WriteLine(byteValue3);
+      // The example displays the following output:
+      //          201
+      //          201
+      //          201
+      // </SnippetByte>
+   }
+
+   private static void AssignByteWithSeparator()
+   {
+      // <SnippetByteS>
+      byte byteValue3 = 0b1100_1001;
+      Console.WriteLine(byteValue3);
+      // The example displays the following output:
+      //          201
+      // </SnippetByteS>
+   }
+
+   private static void AssignShort()
+   {
+      // <SnippetShort>
+      short shortValue1 = 1034;
+      Console.WriteLine(shortValue1);
+      
+      short shortValue2 = 0x040A;
+      Console.WriteLine(shortValue2);
+      
+      short shortValue3 = 0b010000001010;
```

Comment:
I'd recommend adding the _ separator in all the binary constants. It does improve readability